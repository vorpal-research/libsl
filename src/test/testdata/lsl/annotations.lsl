libsl "1.0.0";
library simple;

typealias Int=int32;

automaton A : Int {
   fun f(@annotation param: Int);

   fun g(@anno(1, "12") param: Int)
}
