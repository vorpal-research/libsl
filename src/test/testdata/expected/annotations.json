{
  "name": "simple",
  "lslVersion": "1.0.0",
  "types": [
    {
      "name": "Int",
      "isPointer": false,
      "kind": "alias",
      "originalType": {
        "name": "int32",
        "isPointer": false,
        "kind": "int32",
        "capacity": "INT32"
      }
    }
  ],
  "automata": [
    {
      "name": "A",
      "type": "Int",
      "constructorVariables": [],
      "variables": [],
      "states": [],
      "shifts": [],
      "functions": [
        {
          "name": "f",
          "automaton": "A",
          "target": "A",
          "hasBody": false,
          "args": [
            {
              "name": "param",
              "type": "Int",
              "annotation": {
                "name": "annotation",
                "args": []
              }
            }
          ],
          "contracts": [],
          "statements": []
        },
        {
          "name": "g",
          "automaton": "A",
          "target": "A",
          "hasBody": false,
          "args": [
            {
              "name": "param",
              "type": "Int",
              "annotation": {
                "name": "anno",
                "args": [
                  {
                    "kind": "integer",
                    "value": 1
                  },
                  {
                    "kind": "string",
                    "value": "12"
                  }
                ]
              }
            }
          ],
          "contracts": [],
          "statements": []
        }
      ]
    }
  ]
}